### PR TITLE
[JAX] Support various implementations of RoPE.

### PR DESCRIPTION
**Pull request by mingxu1067**
1. Support various pairing approaches of coordinates in RoPE. `alternate` is to pair index `i` with `i + d/2`
        , `d` is the hidden dimension. 'consecutive' pairs index `i` with `i + 1`.

### CODE REVIEW

File: transformer_engine/jax/flax/transformer.py [hunk 1]
```diff
@@ -248,15 +251,51 @@ def rotary_pos_emb(x: Array, windows: Tuple[int, int], transpose_batch_sequence:
...
+        )
+
+        sign = jnp.sign(jnp.mod(jnp.arange(embed_dim, dtype=jnp.int32), 2) - 0.5)
+
+        output = x * cos + x_shifted * sin * sign
+        return output
+
+    def canonicalize_group_method(gm):
+        canonicalized_gm = gm.lower().strip().replace('-', '').replace('_', '')
+        assert canonicalized_gm in ['consecutive', 'alternate']
```
Comment (denera) [inline transformer_engine/jax/flax/transformer.py:291]: This probably needs a helpful error message like `Invalid relative positional embedding method: must be either 'alternate' or 'consecutive'.`

**Review by denera — APPROVED** [review]
LGTM! Just a minor comment on an (optional) error message for bad RoPE method.

**Review by nouiz — COMMENTED** [review]
LGMT

**Comment by denera** [comment]
@nouiz Do we need JET CI for this?

**Comment by nouiz** [comment]
I think so. We discovered this issue as the current Rope wasn't used by JET CI if I understood correctly. So that would be the first CI run that use it.
Do we need to update the script to make use of it, or it would be automatic?

File: transformer_engine/jax/flax/transformer.py (hunk 1 above)
Comment (mingxu1067) [reply transformer_engine/jax/flax/transformer.py]: Added.

**Comment by zlsh80826** [comment]
Hi @mingxu1067, I found this PR cause functional regression on the LLaMA model. We can discuss offline in the next week.

**Comment by denera** [comment]
@mingxu1067 Is this tested with Llama and ready to merge?

**Comment by mingxu1067** [comment]
@denera @nouiz @zlsh80826 
Anna verified that this PR could get the expected accuracy on llama2-7B.
Kindly help review this PR and merge. Thank you.

**Comment by denera** [comment]
@mingxu1067 LGTM! Thanks for seeing this through.
